spec/lib/services/api/metric_rollups_service_spec.rb

```diff
@@ -0,0 +1,19 @@
+describe Api::MetricRollupsService do
+  before do
+    allow(Settings.api).to receive(:metrics_default_limit).and_return(1000)
+  end
+
+  it 'validates that all of the parameters are present' do
```

Comment:
@imtayadeway its single responsibility is not about validation. I just felt that it was better to test that the validation is done here rather than add individual tests for the various subcollections. The various subcollections do test the other functionality though. I can alter the tests.